snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -566,168 +555,128 @@ func (b *Bootstrapper) markUnavailable(nodeID ids.NodeID) {
+	ctx context.Context,
+	blk snowman.Block,
+	ancestors map[ids.ID]snowman.Block,
+) error {
+	lastAcceptedHeight, err := b.getLastAcceptedHeight(ctx)
+	if err != nil {
+		return err
+	}
 
-		// If this block is going to be accepted, make sure to update the
-		// tipHeight for logging
-		if blkHeight > b.tipHeight {
-			b.tipHeight = blkHeight
-		}
+	numPreviouslyFetched := b.tree.Len()
 
-		pushed, err := b.Blocked.Push(ctx, &blockJob{
-			log:         b.Ctx.Log,
-			numAccepted: b.numAccepted,
-			blk:         blk,
-			vm:          b.VM,
-		})
-		if err != nil {
-			return err
-		}
+	batch := b.DB.NewBatch()
+	missingBlockID, foundNewMissingID, err := process(
+		batch,
+		b.tree,
+		b.missingBlockIDs,
+		lastAcceptedHeight,
+		blk,
+		ancestors,
+	)
+	if err != nil {
+		return err
+	}
 
-		if !pushed {
-			// We can stop traversing, as we have reached a block that we
-			// previously pushed onto the jobs queue
-			if err := b.Blocked.Commit(); err != nil {
-				return err
-			}
-			return b.tryStartExecuting(ctx)
-		}
+	// Update metrics and log statuses
+	{
+		numFetched := b.tree.Len()
+		b.numFetched.Add(float64(b.tree.Len() - numPreviouslyFetched))
 
-		// We added a new block to the queue, so track that it was fetched
-		b.numFetched.Inc()
+		height := blk.Height()
+		b.tipHeight = max(b.tipHeight, height)
 
-		// Periodically log progress
-		blocksFetchedSoFar := b.Blocked.Jobs.PendingJobs()
-		if blocksFetchedSoFar%statusUpdateFrequency == 0 {
+		if numFetched%statusUpdateFrequency == 0 {
 			totalBlocksToFetch := b.tipHeight - b.startingHeight
 			eta := timer.EstimateETA(
 				b.startTime,
-				blocksFetchedSoFar-b.initiallyFetched, // Number of blocks we have fetched during this run
+				numFetched-b.initiallyFetched,         // Number of blocks we have fetched during this run
 				totalBlocksToFetch-b.initiallyFetched, // Number of blocks we expect to fetch during this run
 			)
 			b.fetchETA.Set(float64(eta))
 
 			if !b.restarted {
 				b.Ctx.Log.Info("fetching blocks",
-					zap.Uint64("numFetchedBlocks", blocksFetchedSoFar),
+					zap.Uint64("numFetchedBlocks", numFetched),
 					zap.Uint64("numTotalBlocks", totalBlocksToFetch),
 					zap.Duration("eta", eta),
 				)
 			} else {
 				b.Ctx.Log.Debug("fetching blocks",
-					zap.Uint64("numFetchedBlocks", blocksFetchedSoFar),
+					zap.Uint64("numFetchedBlocks", numFetched),
 					zap.Uint64("numTotalBlocks", totalBlocksToFetch),
 					zap.Duration("eta", eta),
 				)
 			}
 		}
+	}
 
-		// Attempt to traverse to the next block
-		parentID := blk.Parent()
-
-		// First check if the parent is in the processing blocks set
-		parent, ok := processingBlocks[parentID]
-		if ok {
-			blk = parent
-			continue
-		}
-
-		// If the parent is not available in processing blocks, attempt to get
-		// the block from the vm
-		parent, err = b.VM.GetBlock(ctx, parentID)
-		if err == nil {
-			blk = parent
-			continue
-		}
-		// TODO: report errors that aren't `database.ErrNotFound`
```

Comment:
We technically didn't _need_ to call this here... But it avoids requesting the last accepted block over the p2p.

We now handle this in a more clean way by tracking the last accepted height.